R/brreg.R

```diff
@@ -25,15 +25,51 @@
 #' df <- get_entity(c(974760843, 971524960))
 #'
 get_entity <- function(entity, raw_response = FALSE) {
-  dl <- lapply(entity, function(x) get_entity_safe(x, raw_response))
-  if (raw_response) return(dl)
+  dl <- lapply(entity, function(x)
+    get_brreg_safe(x, type = 'enheter', raw_response))
+  if (raw_response) return(invisible(dl))
   do.call('rbind', dl)
 }
 
+#' Get roles
+#'
+#' Fetch information about entity roles from the
+#' _Central Coordinating Register for Legal Entities_
+#' (Enhetsregisteret) based on the organisation number.
+#'
+#' @details
+#' See the
+#' [API documentation](https://data.brreg.no/enhetsregisteret/api/docs/index.html)
+#' for further details (in Norwegian only).
+#'
+#' @param entity A Norwegian organisation number.
+#' @inheritParams get_entity
+#' @return list
+#' @export
+#' @examples
+#' # Get roles for a single entity
+#' res <- get_roles(974760843)
+#'
+#' # Get roles for a mulitiple enities
+#' res <- get_roles(c(974760843, 971524960))
+#'
+get_roles <- function(entity, raw_response = FALSE) {
+  dl <- lapply(entity, function(x)
+    get_brreg_safe(x, type = 'roller', raw_response))
+  if (raw_response) return(invisible(dl))
+  dl <- list(
+    persons = do.call('rbind', lapply(dl, function(x) x$persons)), #purrr::map_df(dl, function(x) x$persons),
+    entities = do.call('rbind', lapply(dl, function(x) x$entities)) # purrr::map_df(dl, function(x) x$entities)
```

Comment:
If performance is an issue, `pryr::rbind.fill()` is usually the most performant, but `do.call()` should suffice in this case?

Reply:
I don't think performance is an issue _for now_, and would prefer that we rely on base R when possible. One reason I considered `{purrr}` is because its map functions are generally good at forcing the type of output you want as well handling errors and NULL values (which is more of safety argument then a performance one). 

But looking forward to benchmark and profile different options as we are moving forward with the package developmment.